browser/ui/webui/ai_chat/ai_chat_ui_page_handler.cc

```diff
@@ -441,19 +454,20 @@ void AIChatUIPageHandler::GetFaviconImageData(
       &favicon_task_tracker_);
 }
 
-absl::optional<mojom::SiteInfo> AIChatUIPageHandler::BuildSiteInfo() {
-  if (active_chat_tab_helper_ && active_chat_tab_helper_->HasPageContent() ==
-                                     PageContentAssociation::HAS_CONTENT) {
-    mojom::SiteInfo site_info;
+mojom::SiteInfo AIChatUIPageHandler::BuildSiteInfo() {
```

Comment:
I'm glad this is a little simpler, but would it make more sense to finally remove the disjointed events and put BuildSiteInfo in ConversationDriver and have that directly fire `OnSiteInfoChanged` instead of a similarly named event. As it is it seems like we're unneccessarily rewriting events and property names from ConversationDriver through AIChatPageHandler for not much reason.

Reply:
moved BuildSiteInfo to ConversationDriver